lib/purchases_flutter.dart

```diff
@@ -269,32 +273,39 @@ class Purchases {
   /// [productIdentifier] The product identifier of the product you want to
   /// purchase.
   ///
-  /// [upgradeInfo] Android only. Optional UpgradeInfo you wish to upgrade from
+  /// [upgradeInfo] Android and Google Play only. Optional UpgradeInfo you wish to upgrade from
   /// containing the oldSKU and the optional prorationMode.
   ///
+  /// [googleProductChangeInfo] Android and Google Play only. Optional GoogleProductChangeInfo you wish to
+  /// change from containing the googleOldProductIdentifer and the
+  /// optional prorationMode.
+  ///
   /// [type] If the product is an Android INAPP, this needs to be
   /// PurchaseType.INAPP otherwise the product won't be found.
   /// PurchaseType.Subs by default. This parameter only has effect in Android.
   ///
-  /// [isPersonalizedPrice] Android only. Optional isPersonalizedPrice indicates
+  /// [googleIsPersonalizedPrice] Android and Google Play only. Optional isPersonalizedPrice indicates
   /// personalized pricing on products available for purchase in the EU.
   /// For compliance with EU regulations. User will see "This price has been
   /// customize for you" in the purchase dialog when true.
   /// See https://developer.android.com/google/play/billing/integrate#personalized-price
   /// for more info.
   static Future<CustomerInfo> purchaseProduct(
     String productIdentifier, {
-    UpgradeInfo? upgradeInfo,
+    @Deprecated('Use GoogleProductChangeInfo') UpgradeInfo? upgradeInfo,
     PurchaseType type = PurchaseType.subs,
-    bool? isPersonalizedPrice,
+    GoogleProductChangeInfo? googleProductChangeInfo,
+    bool? googleIsPersonalizedPrice,
   }) async {
-    final prorationMode = upgradeInfo?.prorationMode;
+    final prorationMode =
+        googleProductChangeInfo?.prorationMode ?? upgradeInfo?.prorationMode;
     final customerInfo = await _invokeReturningCustomerInfo('purchaseProduct', {
       'productIdentifier': productIdentifier,
-      'oldSKU': upgradeInfo?.oldSKU,
-      'prorationMode': prorationMode?.index,
       'type': describeEnum(type),
-      'isPersonalizedPrice': isPersonalizedPrice,
+      'googleOldProductIdentifier':
+          googleProductChangeInfo?.oldProductIdentifier ?? upgradeInfo?.oldSKU,
+      'googleProrationMode': prorationMode?.index,
```

Comment:
should we still pass through the deprecated one somewhere?

Reply:
jweoifjapwioefjaiowpef... this is based on index which is going to be bad once we add new ones to this 🤦‍♂️ 

I need to fix this and also add unit tests around this